pkg/controller/composite/controller.go

```diff
@@ -304,6 +312,14 @@ func (pc *parentController) processNextWorkItem() bool {
 }
 
 func (pc *parentController) enqueueParentObject(obj interface{}) {
+	// If the parent doesn't match our selector, and it doesn't have our
+	// finalizer, we don't care about it.
+	if parent, ok := obj.(*unstructured.Unstructured); ok {
+		if !controllerutil.ContainsFinalizer(parent, pc.finalizer.Name) && pc.parentSelector != nil && !pc.parentSelector.Matches(labels.Set(parent.GetLabels())) {
```

Comment:
query: `parent.GetLabels()` Does this imply all the labels present on the Parent needs to match to proceed further?
I think it will be nice to update the PR body with a sample YAML of now things will work. Probably you have already thought of doc updates.

Reply:
nit: Does it makes sense to have following condition as a dedicated function in controllerutil? I find this getting repeated across multiple files.
```go
pc.parentSelector != nil && !pc.parentSelector.Matches(labels.Set(parent.GetLabels())
```

Reply:
@AmitKumarDas no,  just the selector need to match given set, so if selector is `app=aa`  and labels set is `{app=aa, other=bb}` it will match

Reply:
also, updated docs and example in `configmappropagation`